landlab/ca/cfuncs.pyx

```diff
@@ -187,23 +178,23 @@ cdef int current_link_state(
 @cython.boundscheck(True)
 @cython.wraparound(False)
 cpdef update_link_states_and_transitions(
-    np.ndarray[DTYPE_INT_t, ndim=1] active_links,
-    np.ndarray[DTYPE_INT_t, ndim=1] node_state,
-    np.ndarray[DTYPE_INT_t, ndim=1] node_at_link_tail,
-    np.ndarray[DTYPE_INT_t, ndim=1] node_at_link_head,
-    np.ndarray[DTYPE_INT8_t, ndim=1] link_orientation,
-    np.ndarray[DTYPE_INT8_t, ndim=1] bnd_lnk,
-    np.ndarray[DTYPE_INT_t, ndim=1] link_state,
-    np.ndarray[DTYPE_INT_t, ndim=1] n_xn,
+    const id_t [:] active_links,
```

Comment:
The `const` should allow the compiler so make some optimizations and speed up the code. I, however, like it because it makes the code a little more readable and alerts me when I accidentally try to  modify an array that I'm not supposed to modify.